src/notation/internal/notationinteraction.cpp

```diff
@@ -1949,12 +1950,16 @@ void NotationInteraction::editText(QKeyEvent* event)
     m_textEditData.key = event->key();
     m_textEditData.modifiers = event->modifiers();
     m_textEditData.s = event->text();
+    startEdit();
     if (m_textEditData.element->edit(m_textEditData)) {
         event->accept();
+        apply();
+    } else {
+        m_undoStack->rollbackChanges();
+    }
+    if (!wasEditingText) {
+        m_textEditData.element = nullptr;
```

Comment:
Fixes a bug I'd introduced a while back, where `m_textEditData.element` could be left referring to a deleted element after undo etc., caused very weird app crashes that MSVC debugger couldn't even catch.